File: icms_project/backend/app/models/medication.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base

class Medication(Base):
    __tablename__ = "medications"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    dosage = Column(String, nullable=False)
    frequency = Column(String, nullable=False)
    start_date = Column(DateTime, nullable=False)
    end_date = Column(DateTime, nullable=False)
    instructions = Column(String)
    patient_id = Column(Integer, ForeignKey("patients.id"), nullable=False)
    prescribed_by = Column(Integer, ForeignKey("users.id"), nullable=False)
    active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    patient = relationship("Patient", back_populates="medications")
    doctor = relationship("User", back_populates="prescribed_medications")
    reminders = relationship("MedicationReminder", back_populates="medication")

class MedicationReminder(Base):
    __tablename__ = "medication_reminders"

    id = Column(Integer, primary_key=True, index=True)
    medication_id = Column(Integer, ForeignKey("medications.id"), nullable=False)
    patient_id = Column(Integer, ForeignKey("patients.id"), nullable=False)
    scheduled_time = Column(DateTime, nullable=False)
    status = Column(String, nullable=False)  # PENDING, SENT, ACKNOWLEDGED, MISSED
    type = Column(String, nullable=False)    # SMS, EMAIL, PUSH
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    medication = relationship("Medication", back_populates="reminders")
    patient = relationship("Patient", back_populates="medication_reminders")

File: icms_project/frontend/icms-frontend/src/types/medication.ts
export interface Medication {
  id: number;
  name: string;
  dosage: string;
  frequency: string;
  startDate: string;
  endDate: string;
  instructions: string;
  patientId: number;
  prescribedBy: number;
  active: boolean;
}

export interface MedicationReminder {
  id: number;
  medicationId: number;
  patientId: number;
  scheduledTime: string;
  status: 'PENDING' | 'SENT' | 'ACKNOWLEDGED' | 'MISSED';
  type: 'SMS' | 'EMAIL' | 'PUSH';
}

export interface CreateMedicationDTO {
  name: string;
  dosage: string;
  frequency: string;
  startDate: string;
  endDate: string;
  instructions: string;
  patientId: number;
  prescribedBy: number;
  reminderPreferences: {
    email: boolean;
    sms: boolean;
    push: boolean;
  };
}

export interface UpdateMedicationDTO extends Partial<CreateMedicationDTO> {
  id: number;
}
